mod/file/actions/file/upload.php

```diff
@@ -68,6 +68,10 @@
 		// user blanked title, but we need one
 		$title = $file->title;
 	}
+
+	if ($entity->access_id !== $access_id) {
```

Comment:
These changes would make no sense in 2.0. They only make sense for icons handled by the new API. What happens here is we update icons' modified times, so all requests to previously generated URLs will fail due to HMAC mismatch.